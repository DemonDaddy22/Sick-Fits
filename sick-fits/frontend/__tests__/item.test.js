import { shallow } from 'enzyme';
import ItemComponent from '../components/Item';
import toJSON from 'enzyme-to-json';

const fakeItem = {
    id: 'foo123',
    title: 'Foobar',
    description: 'This is foobar',
    price: 2000,
    image: 'foo.jpg',
    largeImage: 'bar.jpg'
};

describe('<Item />', () => {
    it('should render item and take snapshot', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});

/*
describe('<Item />', () => {
    it('should render title and price tag correctly', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem} />);
        // console.log(wrapper.debug());
        const PriceTag = wrapper.find('PriceTag');
        expect(PriceTag.dive().text()).toBe('$20');
        expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
    });

    it('should render image correctly', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem} />);
        const img = wrapper.find('img');
        expect(img.props().src).toBe(fakeItem.image);
        expect(img.props().alt).toBe(fakeItem.title);
    });

    it('should render buttons correctly', () => {
        const wrapper = shallow(<ItemComponent item={fakeItem} />);
        const buttonList = wrapper.find('.buttonList');
        expect(buttonList.children()).toHaveLength(3);
        // console.log(buttonList.debug());
        expect(buttonList.find('Link')).toBeTruthy();
        expect(buttonList.find('AddToCart')).toBeTruthy();
        expect(buttonList.find('DeleteItem').exists()).toBe(true);
    })
});
*/