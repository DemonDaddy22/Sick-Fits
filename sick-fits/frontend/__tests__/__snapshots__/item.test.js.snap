// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> should render item and take snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Foobar"
    src="foo.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "foo123",
          },
        }
      }
    >
      <a>
        Foobar
      </a>
    </Link>
  </Title>
  <PriceTag>
    $20
  </PriceTag>
  <p>
    This is foobar
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "foo123",
          },
        }
      }
    >
      <a>
        Edit
      </a>
    </Link>
    <AddToCart
      id="foo123"
    />
    <DeleteItem
      id="foo123"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
